### PR TITLE
Create Shareable presenter

### PR DESCRIPTION
**Pull request by bevanloon**
Share links will be required by multiple formats.
Refactoring the code into a Shareable presenter will allow other formats
to include the functionality easily.

### CODE REVIEW

**Comment by bevanloon** [comment]
Would be good to get some hints on how to test the Shareable module we've created - what we have feels a little clumsy and relies on knowledge of what the private methods in Shareable are doing.

**Comment by fofr** [comment]
What formats will need this other than consultations?

**Comment by bevanloon** [comment]
World location news articles

File: app/views/content_items/consultation.html.erb [hunk 1]
```diff
@@ -231,18 +231,7 @@
...
-          target: "_blank",
-          class: "share-button-link js-share-facebook" do %>
-        <span class="share-button share-button-facebook"></span><span class="visually-hidden">Share on </span>Facebook<% end %>
-
-      <%= link_to @content_item.twitter_share_url,
-          target: "_blank",
-          class: "share-button-link js-share-twitter" do %>
-        <span class="share-button share-button-twitter"></span><span class="visually-hidden">Share on </span>Twitter<% end %>
-    </div>
+    <%= render 'shared/share_icons', share_urls: @content_item.share_urls %>
```
Comment (fofr) [inline app/views/content_items/consultation.html.erb]: Could we call the partial `share_buttons`? The partial provides more than icons.

**Comment by danielroseman** [comment]
Should this perhaps be a shared component?

**Comment by bevanloon** [comment]
@danielroseman - we did think about that - we weren't sure if that was necessary at this point given only two formats are using it so far so we put this together in the meantime....

happy to make it a shared component if that's a better way to go?

**Comment by danielroseman** [comment]
It was just a suggestion, happy for it to be here for now and extracted later if necessary.

File: app/views/content_items/consultation.html.erb (hunk 1 above)
Comment (bevanloon) [inline app/views/content_items/consultation.html.erb]: Changed in 1f7d1c5f998ebc95904523c6b53cbe49d58c1c32

**Comment by fofr** [comment]
Nothing outside of this app is using the share button pattern yet, my preference is for it to stay here until it needs to be abstracted a level higher.

**Pull request merged**